src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -301,6 +311,102 @@ private void AssertLicenseLimits(string type, ServerStore serverStore, DatabaseR
                     return;
 
                 throw new LicenseLimitException(LimitType.DataArchival, "Your license doesn't support using the data archival feature.");
+            case nameof(UpdatePullReplicationAsSinkCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60101) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasPullReplicationAsSink)
+                    return;
+
+                throw new LicenseLimitException(LimitType.PullReplicationAsSink, "Your license doesn't support adding Sink Replication feature.");
+            case nameof(UpdatePullReplicationAsHubCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60101) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasPullReplicationAsHub)
+                    return;
+
+                throw new LicenseLimitException(LimitType.PullReplicationAsHub, "Your license doesn't support adding Hub Replication feature.");
+
+            case nameof(UpdateExternalReplicationCommand):
+                if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60101) == false)
+                    return;
+
+                if (serverStore.LicenseManager.LicenseStatus.HasExternalReplication)
+                    if (serverStore.LicenseManager.LicenseStatus.HasDelayedExternalReplication)
+                        return;
+
+                if (databaseRecord.ExternalReplications.Last().DelayReplicationFor == TimeSpan.Zero)
```

Comment:
why `Last` ?

Reply:
This is old code and doesn't exist in the fix. After the fix in v5.4 is done and merged, I will fix v6.0